Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -49,52 +52,291 @@ public FunctionManager(IEnvironment environment, ITraceFactory traceFactory)
         public bool IsFunctionEnabled()
         {
             // this should read appSettings instead
-            var hostJson = Path.Combine(_environment.WebRootPath, HostJsonFile);
+            var hostJson = Path.Combine(_environment.WebRootPath, Constants.FunctionsHostConfigFile);
             return FileSystemHelpers.FileExists(hostJson);
         }
 
-        public JArray GetTriggerInputs(ITracer tracer)
+        public async Task<JArray> GetTriggerInputs(ITracer tracer)
         {
             JArray inputs = new JArray();
-            foreach (var functionJson in EnumerateFunctionFiles())
+            foreach (var functionJson in await ListFunctionsConfig())
             {
                 try
                 {
-                    var json = JObject.Parse(FileSystemHelpers.ReadAllText(functionJson));
-                    var binding = json.Value<JObject>("bindings");
```

Comment:
This would be a great place / time to address https://github.com/projectkudu/kudu/issues/1891 while you're in here :)